Add Dev Container support

MiddelkoopT opened the pull request:
## Description of the Pull Request (PR):

Add preliminary Development Containers support for VSC (https://containers.dev/)
Based on a Rocky9 container and includes the following features.
 * common-utils (creates vscode user)
  * go toolchain (latest)
  * makefile support
 
Lightly tested but will build and `make test` in a Dev Container.  This is a dev feature so no CHANGELOG added.

## Reviewer  checklist

The reviewer checks the following items before merging the PR.

- [x] The PR is based on the appropriate branch (typically [main](https://github.com/warewulf/warewulf/tree/main/userdocs))
- [x] All commits are "Signed off" (e.g., using `git commit --signoff`) in agreement to the [DCO](DCO.txt)
- [x] The [CHANGELOG](https://github.com/warewulf/warewulf/blob/main/CHANGELOG.md) has been updated, if necessary, and under the correct release heading
- [x] The [userdocs](https://github.com/warewulf/warewulf/tree/main/userdocs) have been updated, if necessary
- [x] The submitter is listed in the [contributors file](https://github.com/warewulf/warewulf/blob/main/CONTRIBUTORS.md)
- [ ] The test suite has been updated, if necessary

MiddelkoopT commented:
Added Changelog.

anderbubble reviewed (changes requested):
@MiddelkoopT thanks for the contribution!

Could you draft some documentation on how to use this for the userdocs? (With upstream links for more information; you don't need to document all of VSCode here, of course.)

Let me know if you need help getting it into the docs.

MiddelkoopT commented:
> @MiddelkoopT thanks for the contribution!
> 
> Could you draft some documentation on how to use this for the userdocs? (With upstream links for more information; you don't need to document all of VSCode here, of course.)
> 
> Let me know if you need help getting it into the docs.

I have added some docs.  Let me know if that is what you where thinking.  I also added build-deps to the container to build the documentation as well.
